### PR TITLE
Removed deprecated methods from OperationService

### PR DESCRIPTION
**Pull request by pveentjer**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/spi/OperationService.java [hunk 1]
```diff
@@ -112,19 +46,6 @@
...
-     * method, otherwise {@link #executeOperation(Operation)} should be used.
-     *
-     * @param op the operation to check.
-     * @return true if the operation is allowed to run on the calling thread, false otherwise.
-     *
-     * @deprecated since 3.5 since not needed anymore.
-     */
-    @Deprecated
-    boolean isAllowedToRunOnCallingThread(Operation op);
-
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/spi/OperationService.java]: What about the last deprecated method?
`boolean send(Response response, Address target);`

Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/OperationService.java]: Will be removed in a different pr (removal of response objects).

**Comment by tombujok** [comment]
:+1:

**Comment by jerrinot** [comment]
:+1:

File: hazelcast/src/main/java/com/hazelcast/spi/OperationService.java (hunk 1 above)
Comment (mdogan) [inline hazelcast/src/main/java/com/hazelcast/spi/OperationService.java]: btw, just a reminder... `isAllowedToRunOnCallingThread()` is used by EE IMap and ICache.

Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/spi/OperationService.java]: Is this a showstopper for this PR? It already has two reviewers, but if we use this method internally, it cannot be removed. Or is there an alternative which should be used instead?
